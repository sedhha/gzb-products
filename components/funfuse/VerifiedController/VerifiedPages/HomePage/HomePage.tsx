import AppWrapper from '@/components/funfuse/hoc/AppWrapper';
import React from 'react';

export default function HomePage() {
  return (
    <AppWrapper
      childComponent={<div>This is Home Route</div>}
      topNavBarprops={{ headerText: 'Home' }}
    />
  );
}
