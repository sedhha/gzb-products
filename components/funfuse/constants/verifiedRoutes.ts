export const verifiedRoutes = {
  HOME_ROUTE: 'home',
  MESSAGES_ROUTE: 'messages',
  MENTORSHIP_ROUTE: 'mentorship',
  EVENTS_ROUTE: 'events',
  CONNECTS_ROUTE: 'connects',
};

export const navBarHeaders = {
  HOME_ROUTE: 'Home',
  MESSAGES_ROUTE: 'Messages',
  MENTORSHIP_ROUTE: 'Mentorship',
  EVENTS_ROUTE: 'Events',
  CONNECTS_ROUTE: 'Connects',
  UNKNOWN_ROUTE: '404',
  PROFILE_ROUTE: 'Profile',
};
export const navBarItems = {
  HOME_ROUTE: 'funfuse-icons-home',
  MESSAGES_ROUTE: 'funfuse-icons-messages',
  MENTORSHIP_ROUTE: 'funfuse-icons-mentorship',
  EVENTS_ROUTE: 'funfuse-icons-events',
  CONNECTS_ROUTE: 'funfuse-icons-connects',
};

export const genericRoutes = {
  UNVERIFIED: 'unverified-profile',
  PROFILE_ROUTE: 'profile',
  VERIFY_EMAIL_ROUTE: 'verify-email',
  VIEW_ROUTE: 'view',
};
