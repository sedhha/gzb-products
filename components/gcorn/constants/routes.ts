export const routeProps = [
  {
    id: 0,
    buttonText: 'Sign In',
    routeLocation: 'login',
  },
  {
    id: 1,
    buttonText: 'Create An Account',
    routeLocation: 'register',
  },
];
