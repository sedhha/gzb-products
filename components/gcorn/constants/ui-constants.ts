export const ariaLabels = {
  HOME_PAGE_ARIA_LABEL_DESCRIPTOR: 'GCorn-HomePage-',
};
