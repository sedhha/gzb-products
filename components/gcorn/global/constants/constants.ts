const AppConstants = {
  // Themes
  THEME_GREEN: '#00b050',
  //Routes
  BASE_ROUTE: '/gcorn',
  ROUTE_HOME: '/',
  ROUTE_LOGIN: '/login',
};

export default AppConstants;
