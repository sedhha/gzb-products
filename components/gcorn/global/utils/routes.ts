import AppConstants from '../constants/constants';

export const routeSynthesizer = (location: string) =>
  `${AppConstants.BASE_ROUTE}/${location}`;
