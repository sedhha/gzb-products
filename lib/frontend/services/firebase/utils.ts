import { DataSnapshot } from 'firebase/database';
export const converSnapShotIntoArray = (snapshot: DataSnapshot): string[] => {
  if (snapshot.exists()) {
    const result = [] as string[];
    snapshot.forEach((element) => {
      result.push(element.val());
    });
    return result;
  }
  return [];
};
