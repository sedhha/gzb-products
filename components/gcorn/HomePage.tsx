import React from 'react';
import Head from 'next/head';
import classes from './Home.module.scss';
import Image from 'next/image';
import OutlineButton from '../common/Buttons/OutlineButton';
import AppConstants from './global/constants/constants';
import { useRouter } from 'next/router';
import { routeSynthesizer } from './global/utils/routes';
import { routeProps } from './constants/routes';
import { ariaLabels } from './constants/ui-constants';

const { HOME_PAGE_ARIA_LABEL_DESCRIPTOR } = ariaLabels;

export default function GCornHomePage() {
  const router = useRouter();
  const onNavigationButtonClick = (path: string) => {
    const routeLocation = routeSynthesizer(path);
    router.push(routeLocation);
  };
  return (
    <React.Fragment>
      <Head>
        <title>gCorn: A Corner for Geeks</title>
        <link rel='icon' href='/gcorn/gcorn-logo.svg' />
      </Head>
      <main className={classes.mainSection}>
        <div className={'image-container'}>
          <Image
            src={'/gcorn/gcorn-logo.svg'}
            alt={'gcorn-logo'}
            layout='fill'
            objectFit='contain'
            priority
          />
        </div>
        <div className={classes.WrapperDiv}>
          <h1 className={classes.Header}>gCorn!</h1>
          <p className={classes.Tagline}>A Corner for Geeks</p>
        </div>
        <br />
        {routeProps.map((route) => {
          return (
            <OutlineButton
              key={route.id}
              outlineColor={AppConstants.THEME_GREEN}
              aria-label={
                HOME_PAGE_ARIA_LABEL_DESCRIPTOR +
                'OUTLINE_BUTTON' +
                route.routeLocation
              }
              onClick={() => onNavigationButtonClick(route.routeLocation)}>
              {route.buttonText}
            </OutlineButton>
          );
        })}
      </main>
    </React.Fragment>
  );
}
