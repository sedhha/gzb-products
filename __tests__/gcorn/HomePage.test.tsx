import HomePage from '@/components/gcorn/HomePage';
import { render, screen, fireEvent } from '@testing-library/react';
import * as NextRouter from 'next/router';
import { routeSynthesizer } from '@/components/gcorn/global/utils/routes';
import { routeProps } from '@/components/gcorn/constants/routes';
import { ariaLabels } from '@/components/gcorn/constants/ui-constants';
const { HOME_PAGE_ARIA_LABEL_DESCRIPTOR } = ariaLabels;

describe('HomePage Test', () => {
  it('should render without throwing an error', () => {
    const { container } = render(<HomePage />);
    expect(container).toMatchSnapshot();
  });

  it('should contain "OUTLINE-BUTTON" section', () => {
    const pushFunction = jest.fn();
    const routerFunction = jest
      .spyOn(NextRouter, 'useRouter')
      .mockImplementation(jest.fn().mockReturnValue({ push: pushFunction }));
    render(<HomePage />);

    routeProps.forEach((element) => {
      const navButton = screen.getByLabelText(
        HOME_PAGE_ARIA_LABEL_DESCRIPTOR +
          'OUTLINE_BUTTON' +
          element.routeLocation
      );
      expect(navButton).toBeInTheDocument();
      fireEvent.click(navButton);
      expect(pushFunction).toHaveBeenCalledWith(
        routeSynthesizer(element.routeLocation)
      );
    });
    routerFunction.mockReset();
  });
});
