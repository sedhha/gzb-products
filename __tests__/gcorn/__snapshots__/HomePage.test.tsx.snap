// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Test should render without throwing an error 1`] = `
<div>
  <main
    class="mainSection"
  >
    <div
      class="image-container"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="gcorn-logo"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/gcorn/gcorn-logo.svg"
          srcset="/gcorn/gcorn-logo.svg 640w, /gcorn/gcorn-logo.svg 750w, /gcorn/gcorn-logo.svg 828w, /gcorn/gcorn-logo.svg 1080w, /gcorn/gcorn-logo.svg 1200w, /gcorn/gcorn-logo.svg 1920w, /gcorn/gcorn-logo.svg 2048w, /gcorn/gcorn-logo.svg 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
      </span>
    </div>
    <div
      class="WrapperDiv"
    >
      <h1
        class="Header"
      >
        gCorn!
      </h1>
      <p
        class="Tagline"
      >
        A Corner for Geeks
      </p>
    </div>
    <br />
    <button
      aria-label="GCorn-HomePage-OUTLINE_BUTTONlogin"
      class="sc-bdvvtL sc-gsDKAQ byMHED gKCaVh"
    >
      Sign In
    </button>
    <button
      aria-label="GCorn-HomePage-OUTLINE_BUTTONregister"
      class="sc-bdvvtL sc-gsDKAQ byMHED gKCaVh"
    >
      Create An Account
    </button>
  </main>
</div>
`;
